src/BurgerIcon.js

```diff
@@ -79,8 +79,16 @@ export default class BurgerIcon extends Component {
         {icon}
         <button
           onClick={this.props.onClick}
-          onMouseOver={() => this.setState({ hover: true })}
-          onMouseOut={() => this.setState({ hover: false })}
+          onMouseOver={() => {
+            this.setState({ hover: true });
+            if (this.props.onIconStateChange)
+              this.props.onIconStateChange({ isMouseIn: true });
```

Comment:
My only suggestion would be changing the name to `onIconHoverChange` to make it more obvious it's just triggered on the hover event.

Reply:
Thanks! Will do that now.

Reply:
Done.

Reply:
Thanks!